src/qml/modules/Shotcut/Controls/GradientMapControl.qml

```diff
@@ -0,0 +1,307 @@
+
+            Shotcut.ColorDialog {
+                id: colorDialog
+
+                title: qsTr("Color #%1").arg(stopIndex + 1)
+                selectedColor: handleRect.color
+                onAccepted: gradientMap._setStopColor(handleRect.stopIndex, String(selectedColor), getPosition())
+            }
+
+            Shotcut.HoverTip {
+                text: qsTr('Color: %1\nClick to select, drag to change position').arg(color)
+            }
+
+            MouseArea {
+                id: stopArea
+                anchors.fill: parent
+                drag.target: parent
+                drag.axis: Drag.XAxis
+                drag.minimumX: ((parent.prev && parent.prev.position) || 0) * gradientFrame.width - width / 2
+                drag.maximumX: ((parent.next && parent.next.position) || 1) * gradientFrame.width - width / 2
+                drag.onActiveChanged: {
+                    if (!drag.active) {
+                        const position = getPosition();
+                        stopPosition.value = position;
+                        gradientMap._setStopColor(handleRect.stopIndex, String(handleRect.color), position);
+                    } else if (stopSpinner.value !== handleRect.stopIndex) {
+                        stopSpinner.value = handleRect.stopIndex;
+                    }
+                }
+                onClicked: {
+                    if (stopSpinner.value !== handleRect.stopIndex) {
+                        stopSpinner.value = handleRect.stopIndex;
+                    }
+                }
+            }
+        }
+    }
+
+    Rectangle {
+        id: gradientFrame
+
+        Layout.columnSpan: 3
+        Layout.fillWidth: true
+        implicitHeight: 20
+        implicitWidth: 200
+        color: "transparent"
+
+        Rectangle {
+            id: gradientRect
+
+            width: parent.width
+            height: parent.height - 6
+            border.color: "gray"
+            border.width: 1
+            radius: 4
+            gradient: Gradient {
+                id: gradientView
+                orientation: Gradient.Horizontal
+            }
+        }
+    }
+
+    Shotcut.DoubleSpinBox {
+        id: stopSpinner
+
+        width: 100
+        value: 0
+        from: 0
+        to: stops.length - 1
+        stepSize: 1
+        onValueChanged: {
+            for (var idx = 0; idx < gradientMap.stops.length; idx++) {
+                var stop = gradientMap.stops[idx];
+            }
+            var stop = gradientMap.stops[value];
+            stopPosition.value = stop.position;
+        }
+    }
+
+    Shotcut.DoubleSpinBox {
+        id: stopPosition
+
+        width: 100
+        value: typeof stops[stopSpinner.value] != 'undefined' ? stops[stopSpinner.value].position : 0
+        decimals: 3
+        from: 0
+        to: 1
+        stepSize: 0.01
+        onValueModified: _setStopColor(stopSpinner.value, stops[stopSpinner.value].color, value)
+    }
+
+    Rectangle {
+        id: stopColor
```

Comment:
This should have a label such as "Color".
 Finally, this can be in one row with the above 2 controls like
`Selected Stop   __1__   Position   __50 %_   Color [ ]`
Does that make sense?

Reply:
I'd rather put the color into its row because I find it more intuitive for it to be like that. I could take a look at the other two, though.